flutter/lib/desktop/pages/desktop_setting_page.dart

```diff
@@ -1678,29 +1678,111 @@ class _DisplayState extends State<_Display> {
+    final edgeScrollEdgeThicknessRange = bind.mainGetOptionRange(key: kOptionEdgeScrollEdgeThickness);
+    final edgeScrollEdgeThickness = double.parse(bind.mainGetUserDefaultOption(key: kOptionEdgeScrollEdgeThickness));
+    final edgeScrollEdgeThicknessTextController = TextEditingController();
+
+    edgeScrollEdgeThicknessTextController.text = edgeScrollEdgeThickness.toInt().toString();
+
+    updateSavedScrollStyle(String value) async {
       await bind.mainSetUserDefaultOption(
-          key: kOptionScrollStyle, value: value);
+        key: kOptionScrollStyle, value: value);
+    }
+
+    onScrollStyleChanged(String value) async {
+      await updateSavedScrollStyle(value);
       setState(() {});
     }
 
-    final groupValue = bind.mainGetUserDefaultOption(key: kOptionScrollStyle);
+    automaticallySwitchToEdgeScroll() async {
+      if (!isOptFixed && (groupValue != kRemoteScrollStyleEdge)) {
+        await updateSavedScrollStyle(kRemoteScrollStyleEdge);
+      }
+    }
+
+    onEdgeScrollEdgeThicknessChanged(double value) async {
+      var stringValue = value.round().toString();
+      await bind.mainSetUserDefaultOption(
+          key: kOptionEdgeScrollEdgeThickness, value: stringValue);
+      setState(() {});
+    }
+
+    onEdgeScrollEdgeThicknessTextSubmitted(String value) async {
+      int? intValue = int.tryParse(edgeScrollEdgeThicknessTextController.text);
+
+      if (intValue != null) {
+        await onEdgeScrollEdgeThicknessChanged(intValue.toDouble());
+      } else {
+        // can't parse what the user typed, just revert it to what's saved
+        setState(() {});
+      }
+    }
+
+    final edgeScrollEdgeThicknessFocusNode = FocusNode();
+
+    edgeScrollEdgeThicknessFocusNode.addListener(
+      () async {
+        if (edgeScrollEdgeThicknessFocusNode.hasFocus) {
+          automaticallySwitchToEdgeScroll();
+        } else {
+          await onEdgeScrollEdgeThicknessTextSubmitted(edgeScrollEdgeThicknessTextController.text);
+        }
+      });
+
     return _Card(title: 'Default Scroll Style', children: [
       _Radio(context,
           value: kRemoteScrollStyleAuto,
           groupValue: groupValue,
           label: 'ScrollAuto',
-          onChanged: isOptFixed ? null : onChanged),
-      _Radio(context,
-          value: kRemoteScrollStyleEdge,
-          groupValue: groupValue,
-          label: 'ScrollEdge',
-          onChanged: isOptFixed ? null : onChanged),
+          onChanged: isOptFixed ? null : onScrollStyleChanged),
       _Radio(context,
           value: kRemoteScrollStyleBar,
           groupValue: groupValue,
           label: 'Scrollbar',
-          onChanged: isOptFixed ? null : onChanged),
+          onChanged: isOptFixed ? null : onScrollStyleChanged),
+      _Radio(context,
+          value: kRemoteScrollStyleEdge,
+          groupValue: groupValue,
+          label: 'ScrollEdge',
+          onChanged: isOptFixed ? null : onScrollStyleChanged),
+      Container(
```

Comment:
Display the slider only when 'ScrollEdge' is selected, follow the custom image quality setting.

Reply:
I like being able to select ScrollEdge by interacting with the slider, but I can switch it.